code_generation/configuration.py

```diff
@@ -88,7 +89,7 @@ def __init__(
         self.available_scopes = set(available_scopes)
         self.available_outputs: Dict[str, QuantitiesStore] = {}
         self.available_shifts: Dict[str, Set[str]] = {}
-        self.global_scope = "global"
+        self.global_scope = global_scope
```

Comment:
If you want to do this you probably also need propagate this everywhere where `"global"` is explicitly used. Like here
https://github.com/KIT-CMS/CROWN/blob/95ea5cb8f6bf33ed5dbfc7ab635587a093527a40/code_generation/producer.py#L149
https://github.com/KIT-CMS/CROWN/blob/95ea5cb8f6bf33ed5dbfc7ab635587a093527a40/code_generation/rules.py#L23
https://github.com/KIT-CMS/CROWN/blob/95ea5cb8f6bf33ed5dbfc7ab635587a093527a40/code_generation/optimizer.py#L82
as just a few examples.

Reply:
The intended change here is to only provide the option to not have a `global` scope at all. A more direct choice for this was implemented in [Link](https://github.com/KIT-CMS/CROWN/pull/286/commits/542ca2cdd09a8dc7cbc11bb865fd47daaaf72752#diff-bac2f5f7803facde8e079f690ad030af05a725fed5d366f97a2f0d41516b2430). As of now, there is no documentation for this feature.

Renaming the global scope is not really needed in my opinion.
It is currently not possible to only have a global scope due to a separate bug that is out of the scope of this PR.
The two cases lead to a very similar outcome (Only having a single scope), but the option to not have a global scope at all leaves more options for the naming of scopes. It also works with the case in which all scopes are treated differently from the very start and no global scope would be used.